pandas/tests/extension/base/ops.py

```diff
@@ -119,6 +119,12 @@ def test_direct_arith_with_series_returns_not_implemented(self, data):
                 "{} does not implement add".format(data.__class__.__name__)
             )
 
+    def test_arith_diff_lengths(self, data, all_arithmetic_operators):
+        op = self.get_op_from_name(all_arithmetic_operators)
+        other = data[:3]
```

Comment:
This test actually wouldn't catch the behavior in #23078, which is caused specifically by comparisons of PeriodIndex with length-1 objects.

Reply:
The issue in #23078 was treated in the function test_comp_op (pandas/tests/indexes/period/test_period.py).